crates/compiler/gen_dev/src/object_builder.rs

```diff
@@ -29,6 +29,28 @@ pub fn build_module<'a, 'r>(
     layout_interner: &'r mut STLayoutInterner<'a>,
     target: &Triple,
     procedures: MutMap<(symbol::Symbol, ProcLayout<'a>), Proc<'a>>,
+) -> Object<'a> {
+    let module_object = build_module_help(env, interns, layout_interner, target, procedures);
+
+    if std::env::var("ROC_DEV_WRITE_OBJ").is_ok() {
+        let module_out = module_object
+            .write()
+            .expect("failed to build output object");
+
+        let file_path = std::env::temp_dir().join("app.o");
+        println!("gen-test object file written to {}", file_path.display());
+        std::fs::write(&file_path, module_out).expect("failed to write object to file");
+    }
```

Comment:
moved here so it is evaluated whenever we use the dev backend (dev tests, repl, or glue with dev)